nifi-frontend/src/main/frontend/libs/shared/src/pipes/sort-by-property.pipe.spec.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SortableBy, SortObjectByPropertyPipe } from './sort-by-property.pipe';
+
+describe('SortObjectByPipe', () => {
+    let pipe: SortObjectByPropertyPipe;
+
+    beforeEach(() => {
+        pipe = new SortObjectByPropertyPipe();
+    });
+
+    it('should sort objects alphabetically by name', () => {
+        const items: SortableBy[] = [
+            { name: 'Banana', value: 'Fruit' },
+            { name: 'Apple', value: 'Fruit' },
+            { name: 'Carrot', value: 'Vegetable' }
+        ];
+        const result = pipe.transform(items);
+        expect(result).toEqual([
+            { name: 'Apple', value: 'Fruit' },
+            { name: 'Banana', value: 'Fruit' },
+            { name: 'Carrot', value: 'Vegetable' }
+        ]);
+    });
+
+    it('should handle case-insensitive sorting', () => {
+        const items: SortableBy[] = [{ name: 'banana' }, { name: 'Apple' }, { name: 'carrot' }];
+        const result = pipe.transform(items);
+        expect(result).toEqual([{ name: 'Apple' }, { name: 'banana' }, { name: 'carrot' }]);
+    });
+
+    it('should return an empty array if input is empty', () => {
+        const items: SortableBy[] = [];
+        const result = pipe.transform(items);
+        expect(result).toEqual([]);
+    });
+
+    it('should handle a single-item array', () => {
+        const items: SortableBy[] = [{ name: 'Apple' }];
+        const result = pipe.transform(items);
+        expect(result).toEqual([{ name: 'Apple' }]);
+    });
+
+    it('should not modify the original array', () => {
+        const items: SortableBy[] = [{ name: 'Banana' }, { name: 'Apple' }];
+        const itemsCopy = [...items];
+        pipe.transform(items);
+        expect(items).not.toEqual(itemsCopy);
+    });
```

Comment:
I'm confused by this test. It suggests that the original array should not get modified however, the underlying transform uses array.sort which sorts array elements in place. In many paces we depend on it modifying the array that is passed in as we don't bother assigning a variable to the result (which is just a reference to the same array).

Reply:
I agree. This test is pretty pointless. I will remove it.